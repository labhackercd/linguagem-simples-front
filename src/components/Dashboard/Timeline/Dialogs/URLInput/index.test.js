import React from 'react';
import { render, screen } from '@testing-library/react';
import {shallow, mount} from "enzyme/build";
import ReactDOM from "react-dom";
import URLInputDialog from './index.js';

it("snapshot should not have differences", () => {
    const component = shallow(<URLInputDialog URLInputDialogOpen={true}/>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

test('Test if Feed renders without crash', () => {
    const div = document.createElement("div")
    ReactDOM.render(<URLInputDialog URLInputDialogOpen={true} />, div)
    ReactDOM.unmountComponentAtNode(div)
});

test('Clicking on buttons doesnt break page', () => {
  const wrapper = mount(<URLInputDialog URLInputDialogOpen={true}
                                        setCustomURL={jest.fn()}
                                        customURL="testURL"
                                        handleDialogStateAction={jest.fn()}/>)
  const textfield = wrapper.find('#url-input').last()
  textfield.instance().value = "test text"
  textfield.simulate('change')
  const submitButton = wrapper.find('#submit-button').last()
  submitButton.simulate('click')
  const deleteButton = wrapper.find('#cancel-button').last()
  deleteButton.simulate('click')
})
