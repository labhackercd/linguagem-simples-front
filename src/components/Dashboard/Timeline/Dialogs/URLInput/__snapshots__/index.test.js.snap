// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should not have differences 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[Function]}
>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      Insira no campo abaixo o link desejado
    </WithStyles(ForwardRef(DialogContentText))>
    <WithStyles(ForwardRef(TextField))
      autoFocus={true}
      fullWidth={true}
      id="name"
      label="Insira aqui o link"
      margin="dense"
      name="customURL"
      onChange={[Function]}
      type="text"
    />
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
    >
      Cancelar
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={true}
      onClick={[Function]}
    >
      Incorporar link
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
